### DISCUSSION TITLE
Problem with osu!lazer's new exclusive full screen mode

### DISCUSSION THREAD

**Discussion by InsertX2k** [ppy/osu#19785] [Q&A] [answered]
Hello, 
So after I got the new osu!lazer update, it seems like I got an issue with the new "exclusive full screen" mode.
The issue is that whenever i switch to the exclusive full screen mode there appears a red word that says "Unable to enter exclusive full screen mode"

My PC Specifications are:
CPU: Intel Core i5 3470
RAM: 12 GB RAM
GPU: Intel HD Graphics 2500 w/ latest GPU Drivers
the issue doesn't seem to be a hardware related, i think it is a bug in osu!lazer as my PC is able to run it at 120fps

**Comment by Doksha** [comment]
I havent ever been able to get exclusive fullscreen to work on integrated graphics, I am 99% sure you need to use a dedicated GPU

**Comment by peppy** [comment, marked as answer]
Hmm, weirdly I can reproduce this on my surface (intel HD 520).

The weird part is that it *is* running in exclusive mode (as confirmed by screen flicker and game bar not displaying properly unless borderless), but the text says that it isn't. @smoogipoo is this a known limitation? If so we probably don't want to display it on intel hardware.

@InsertX2k in the mean time, it's safe to say that your game is likely running in exclusive fullscreen and you have nothing to worry about. You can confirm this by attempting to show game bar (win-g): if it just flashes your screen (or minimises osu! in order to display game bar) then you are running exclusive fullscreen.

**Reply by smoogipoo** [reply]
Please provide logs.

**Reply by peppy** [reply]
Specifically the "fullscreen capability" line?

It shows `Incapable (QUNS_BUSY)`.

**Reply by smoogipoo** [reply]
Right, so I noticed in the original PR that it's only Intel which never enters `QUNS_RUNNING_D3D_FULL_SCREEN`.

At some point, try running https://github.com/smoogipoo/osu-framework/tree/direct-draw-exclusive-fullscreen and giving the logged output.

**Reply by peppy** [reply]
I see many lines stating `0x887600E1`

**Reply by smoogipoo** [reply]
Hmm, that's unfortunate. Seems like there's no way to detect exclusive fullscreen.

**Reply by peppy** [reply]
We probably need to do something about this, as it is incorrectly telling users that exclusive mode is not working, causing confusion..

Tracking at https://github.com/ppy/osu/issues/19961.
